src/server/network/sockets-helper.js

```diff
@@ -0,0 +1,136 @@
+const uWS = require("uWebSockets.js");
+const config = require("../config");
+const log = require("../../log");
+
+const socketsHelper = function() {
+	// Setup socket app
+	const _app = config.network.ssl
+		? uWS.SSLApp({
+			key_file_name: config.uwebsockets.keyFileName,
+			cert_file_name: config.uwebsockets.certFileName,
+			passphrase: config.uwebsockets.passphrase
+		})
+		: uWS.App();
+
+	// Start listening for socket requests
+	let _token;
+	_app.listen(config.uwebsockets.port, (token) => {
+		if (token) {
+			_token = token;
+			log.info(`µWS: Listening to port ${config.uwebsockets.port}`.cyan);
+		} else {
+			log.warn(`µWS: Failed to listen to port ${config.uwebsockets.port}`.cyan);
+		}
+	});
+
+	return {
+		stopListening: function() {
+			uWS.us_listen_socket_close(_token);
+		},
+
+		Socket: function(
```

Comment:
Given that this creates a `Socket` instance and that the chat/gameplay sockets override variables and functions, it may look a lot cleaner if we use proper inheritance for this instead. We only have those two socket types after all.